commands/user.go

```diff
@@ -757,7 +762,15 @@ func listUsersCmdF(c client.Client, command *cobra.Command, args []string) error
 		}
 
 		for _, user := range users {
-			printer.PrintT(tpl, user)
+			switch {
+			case deactivatedUser && user.DeleteAt > 0:
+				showAll = true
+				printer.PrintT(tpl, user)
+			case deactivatedUser && user.DeleteAt <= 0:
+				continue
+			default:
+				printer.PrintT(tpl, user)
+			}
```

Comment:
Getting all users and filtering out doesn't seem to be an elegant solution here for a couple of reasons:
- You don't actually need to update `mmctl`, you can filter out deleted users via `jq` if you pipe the `json` output. 
- This will generate too much data transmitted w/o any value over the network.

I suggest you can add another method to https://github.com/mattermost/mattermost-server/blob/c87e1dd08d1ac90d7569eb4db65b45fd45cd8606/model/client4.go#L1037 such as `GetUsersWithOptions` and use it here to query only inactive users and such. Does it makes sense to you?